### ISSUE TITLE
How to Submit Data without a Form

### ISSUE THREAD

**Issue by jaxon** [gae-init/gae-init#392]
I am trying to commit data parsed from an uploaded file. I thought I had it figured out, but when I run the code, the db is untouched and there are no errors are reported.

Here is a contrived example of the approach I am taking. 

### Model
```   
class Info(model.Base):
  company = ndb.StringProperty(default='',required=True, verbose_name='Company Name')
  type = ndb.StringProperty(default='',required=True, verbose_name='Type')
  FIELDS = {
    'company': fields.String, 
    'type': fields.String,
  }
  FIELDS.update(model.Base.FIELDS)
```

### Code:
```
company = "Google"
type  = "Internet"

info_db = model.Info()
info_db.company = company
info_db.type = internet

form = InfoUpdateForm(obj=info_db)
if form.validate_on_submit():
    form.populate_obj(info_db)
    info_db.put()
```

**Comment by lipis** [comment]
Must me the [`csrf`](https://flask-wtf.readthedocs.org/en/latest/csrf.html)

Try this: 

```
form = InfoUpdateForm(obj=info_db, csrf_enabled=False)
```

**Comment by jaxon** [comment]
No go. Thanks for the tip though.

**Comment by jaxon** [comment]
Basically, I have 10-20 db updated I want to make at any given time. It's too much to manage updating one at a time in a form. What other methods are there for getting data into the db. Data validation would be nice, but not necessary. REST maybe?

**Comment by lipis** [comment]
How are you trying to submit/invoke this request...? You have to do a POST request in order to use the `validate_on_submit()`..

**Comment by lipis** [comment]
Here is a very quick and very dirty example.. https://github.com/lipis/gae-init/commit/dabcf2cde4574530254131159ad08571b24cf228

As you can see.. nothing happens when going directly to: https://jaxon-dot-gae-init.appspot.com/info/update/

But if you go to https://jaxon-dot-gae-init.appspot.com open the console and do the following, it will work:

```
$.post('/info/update/)
$.post('/info/update/?company=SpaceX&category=Aerospace')
```

**Comment by lipis** [comment]
The rest is left as an exercise :)

**Comment by jaxon** [comment]
That was it, no POST. 

I was trying to test REST with CURL, but it seems to block it specifically. The console commands are very handy.

**Comment by lipis** [comment]
so it's solved?

**Comment by jaxon** [comment]
### TL;DR
I was able to reliably post data without a form after switching to a POST request and  removing / ignoring `form.validate_on_submit()`. `csrf` did not appear to be a factor.

### Desription
It worked from the browser console:
```
$.post('/info/update/?company=SpaceX&category=Aerospace')
```
But not in the code:

```
form = InfoUpdateForm(obj=info_db)
if form.validate_on_submit():
    form.populate_obj(info_db)
    info_db.put()
```
Until I removed: `validate_on_submit()..`

Leaving: 
```
form = InfoUpdateForm(obj=info_db)
if form.validate_on_submit():
    form.populate_obj(info_db)
    info_db.put()
```
### More Info
[GitHub](http://github.com)
validate_on_submit()
Checks if form has been submitted and if so runs validate. This is a shortcut, equivalent to form.is_submitted() and form.validate()

**Comment by lipis** [comment]
if you skip the validation.. then you won't have any validation :) like field required, etc..

**Comment by jaxon** [comment]
That's ok at this stage. But I will want validation eventually.

I'll try to replicate this in a project I can share it and hopeful find a working solution which includes validation.

**Comment by lipis** [comment]
I'll close this.. Since you solved your issue.
